fe/fe-core/src/main/java/com/starrocks/planner/OlapScanNode.java

```diff
@@ -481,12 +484,21 @@ public void addScanRangeLocations(Partition partition,
                 replicas = allQueryableReplicas;
             }
 
+            if (!hintsReplicaIds.isEmpty()) {
+                replicas.removeIf(replica -> hintsReplicaIds.contains(replica.getId()));
+                // direct return if no expected replica
+                if (replicas.isEmpty()) {
+                    continue;
```

Comment:
If it's one tablet and  the replicas is empty, is there any problem to return?

Reply:
I have tested. it works